lib/msgServer/index.js

```diff
@@ -480,6 +480,18 @@ function setupDiscovery() {
 
 
 	service.on('error', function (error, service) {
+		if (error.name !== 'CONNECTION_LOSS') {
+			// We ignore connection losses since any persisting connnectivity issues
+			// should be managed by the other event handlers, and connection loss for in-flight requests
+			// will be retried by zookeper.
+			//
+			// See https://github.com/alexguan/node-zookeeper-client/blob/master/index.js#L138 for more details
+			return logger.warning
+				.details('node-zookeeper-client will retry the query; you may ignore')
```

Comment:
We're inside `msgServer` here though, not the zookeeper SD engine. So even if this plays nice with Consul and mDNS (because of error.name), is there not a more appropriate place to put this?